src/Storage/Collection/Relations.php

```diff
@@ -190,7 +190,7 @@ public function getField($fieldname, $biDirectional = false)
     public function incoming($entity)
     {
         return $this->filter(function ($el) use ($entity) {
-            return $el->getTo_contenttype() == (string) $entity->getContenttype();
+            return ($el->getTo_contenttype() == (string) $entity->getContenttype() && $el->getTo_id() == $entity->getId());
```

Comment:
You're right, in this case strict should be fine. Also removed the outer parens.